Trunk Based Development

gsampaio-rh opened the pull request.

mcanoy commented on Jenkinsfile:
```diff
@@ -57,32 +57,14 @@ pipeline {
             }
             when {
-              expression { GIT_BRANCH ==~ /(.*master|.*develop.*|.*feature.*)/ }
+              expression { GIT_BRANCH ==~ /(.*master)/ }
```
We should just assume master. These when blocks aren't needed. This block and the ones below

mcanoy commented:
Seeing a lot of updates here. Is this WIP? If so can you prefix it with WIP: so that someone doesn't go ahead and merge this?

gsampaio-rh commented:
@mcanoy Ready to review. I added two conditional steps:
1. release tag
2. master branch

In order to push it to quay I added a secret entry in the openshift applier.
*The secret was created manually on S11

gsampaio-rh commented:
Quay repo is just a "fake". Once that we have it the correct Quay repo, we can change it.

oybed commented:
@gsampaio-rh looking to either merge or close out this PR. Can you please update to remove the WIP, or if you prefer to close it - let us know. 

BTW: if you choose to keep it, let's update the quay.io to use the `rht-labs` org

gsampaio-rh commented:
@oybed I know that we are moving to GitHub actions but maybe this would still be a "nice to have". Can you review if the quay org is correct?

gsampaio-rh commented:
@oybed Ready to be merged
